lib/src/views/Clock/Clock.tsx

```diff
@@ -167,8 +60,163 @@ export const styles = (theme: Theme) =>
       left: '50%',
       transform: 'translate(-50%, -50%)',
     },
-  });
+    amButton: {
+      zIndex: 24,
+      left: 8,
+      position: 'absolute',
+      bottom: -10,
+    },
+    pmButton: {
+      zIndex: 24,
+      position: 'absolute',
+      bottom: -10,
+      right: 8,
+    },
+    meridiemButtonSelected: {
+      backgroundColor: theme.palette.primary.main,
+      color: theme.palette.primary.contrastText,
+      '&:hover': {
+        backgroundColor: theme.palette.primary.light,
+      },
+    },
+  }),
+  {
+    name: 'MuiPickersClock',
+  }
+);
+
+export const Clock: React.FC<ClockProps> = ({
+  date,
+  onDateChange,
+  ampmInClock = false,
+  value,
+  children: numbersElementsArray,
+  type,
+  ampm,
+  minutesStep,
+  onChange,
+}) => {
+  const classes = useStyles();
+  const isMoving = React.useRef(false);
+  const { meridiemMode, handleMeridiemChange } = useMeridiemMode(date, ampm, onDateChange);
+
+  const isPointerInner = !ampm && type === 'hours' && (value < 1 || value > 12);
+
+  const setTime = (e: any, isFinish = false) => {
+    let { offsetX, offsetY } = e;
+
+    if (typeof offsetX === 'undefined') {
+      const rect = e.target.getBoundingClientRect();
+
+      offsetX = e.changedTouches[0].clientX - rect.left;
+      offsetY = e.changedTouches[0].clientY - rect.top;
+    }
+
+    const value =
+      type === 'seconds' || type === 'minutes'
+        ? getMinutes(offsetX, offsetY, minutesStep)
+        : getHours(offsetX, offsetY, Boolean(ampm));
+
+    onChange(value, isFinish);
+  };
+
+  const handleTouchMove = (e: React.TouchEvent) => {
+    isMoving.current = true;
+    setTime(e);
+  };
+
+  const handleTouchEnd = (e: React.TouchEvent) => {
+    if (isMoving.current) {
+      setTime(e, true);
+      isMoving.current = false;
+    }
+  };
+
+  const handleMouseMove = (e: React.MouseEvent<HTMLDivElement>) => {
+    e.preventDefault();
+    e.stopPropagation();
+    // MouseEvent.which is deprecated, but MouseEvent.buttons is not supported in Safari
+    const isButtonPressed =
+      typeof e.buttons === 'undefined' ? e.nativeEvent.which === 1 : e.buttons === 1;
```

Comment:
I believe `event.button === 0` could work cross platform.

Reply:
We need e.buttons === 1

Reply:
I'm wondering because react-router checks `event.button` to only consider the main click (0).